py/kubeflow/testing/argo_client.py

```diff
@@ -66,7 +66,10 @@ def wait_for_workflows(client, namespace, names,
 
     done = True
     for results in all_results:
-      if results["status"]["phase"] not in ["Failed", "Succeeded"]:
+      # Sometimes it takes a while for the argo controller to populate
+      # the status field of an object.
+      if "status" not in results or "phase" not in results["status"] \
```

Comment:
nit: I think its a little bit easier to read if you use get and return a suitable default
e.g
if results.get("status", {}).get("phase", "") not in ["failed", "Suceeded"]

Reply:
Done